src/main/java/com/uber/profiling/Agent.java

```diff
@@ -33,6 +33,7 @@ public static void premain(final String args, final Instrumentation instrumentat
         System.out.println("Java Agent " + AgentImpl.VERSION + " premain args: " + args);
 
         Arguments arguments = Arguments.parseArgs(args);
+        arguments.processConfigProvider();
```

Comment:
process ConfigFile? as ConfigProvider is an implementation detail, ConfigFile is something faced by the user

Reply:
ConfigProvider is user facing as well, since user could specify configProvider=xxx in the agent argument. Probably I could rename it to runConfigProvider()